libraries/ESPmDNS/src/ESPmDNS.cpp

```diff
@@ -115,7 +113,7 @@ void MDNSResponder::disableArduino(){
 void MDNSResponder::enableWorkstation(esp_interface_t interface){
     char winstance[21+_hostname.length()];
     uint8_t mac[6];
-    esp_wifi_get_mac((wifi_interface_t)interface, mac);
+    esp_base_mac_addr_get(mac);
```

Comment:
How does the ESP_Network_Manager know which MAC address to use here?
The function does get an interface as argument, which is probably for compatibilty reasons. But I truly don't see how the Network Manager or this MDNSResponder knows about which interface is queried here as it doesn't seem to have any members which look like an interface, nor does `ESP_Network_Events`.

So it seems you're only dealing with the base MAC address here and not adding any offset for the WiFi SoftAP, Bluetooth or Ethernet interface.

Reply:
This is subject to change once all interfaces are implemented/reworked. The point of the change currently is to disconnect WiFi from the rest, so things compile.

Reply:
@me-no-dev Is this still valid ?

Reply:
not anymore :)